vulnfeeds/cmd/debian/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/google/osv/vulnfeeds/cves"
+	"github.com/google/osv/vulnfeeds/utility"
+	"github.com/google/osv/vulnfeeds/vulns"
+)
+
+const (
+	debianSecurityTrackerURL = "https://security-tracker.debian.org/tracker/data/json"
+	debianOutputPathDefault  = "parts/debian"
+)
+
+var Logger utility.LoggerWrapper
+
+func main() {
+	var logCleanup func()
+	Logger, logCleanup = utility.CreateLoggerWrapper("debian-osv")
+	defer logCleanup()
+
+	err := os.MkdirAll(debianOutputPathDefault, 0755)
+	if err != nil {
+		Logger.Fatalf("Can't create output path: %s", err)
+	}
+
+	debianData := downloadDebianSecurityTracker()
+	cvePkgInfos := generateDebianSecurityTrackerOSV(debianData)
+	writeToOutput(cvePkgInfos)
+}
+
+// Gets the Debian version number, excluding sid and experimental versions.
+func getDebianReleaseMap() map[string]string {
+	releaseMap := make(map[string]string)
+	res, err := http.Get("https://debian.pages.debian.net/distro-info-data/debian.csv")
+	if err != nil {
+		Logger.Fatalf("Failed to get Debian release info data: %s", err)
+	}
+	defer res.Body.Close()
+
+	reader := csv.NewReader(res.Body)
+	reader.FieldsPerRecord = -1
+	data, err := reader.ReadAll()
+	if err != nil {
+		Logger.Fatalf("Failed to load Debian release info csv: %s", err)
+	}
+
+	versionIndex := -1
+	seriesIndex := -1
+
+	// Get the index number of version and series.
+	for i, col := range data[0] {
+		if col == "version" {
+			versionIndex = i
+		} else if col == "series" {
+			seriesIndex = i
+		}
+	}
+
+	if seriesIndex == -1 || versionIndex == -1 {
+		Logger.Fatalf("Failed to get Debian release info: %s", err)
+	}
+
+	for _, row := range data[1:] {
+		if row[seriesIndex] == "experimental" || row[seriesIndex] == "sid" {
+			continue
+		}
+
+		releaseMap[row[seriesIndex]] = row[versionIndex]
+	}
+
+	return releaseMap
+}
+
+// Convert Debian Security Tracker entries to OSV PackageInfo format.
+func generateDebianSecurityTrackerOSV(debianData DebianSecurityTrackerData) map[string][]vulns.PackageInfo {
+	debianReleaseMap := getDebianReleaseMap()
+	osvPkgInfos := make(map[string][]vulns.PackageInfo)
+	for pkgName, pkg := range debianData {
+		for cveId, cve := range pkg {
+			var pkgInfos []vulns.PackageInfo
+			if value, ok := osvPkgInfos[cveId]; ok {
+				pkgInfos = value
+			}
+
+			for releaseName, release := range cve.Releases {
+				if release.Urgency == "not yet assigned" || release.Urgency == "end-of-life" {
```

Comment:
Can you add some comments here to explain the cases? Ideally also some documentation as well if you can find any.